src/dplasmaaux_hip.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023-2024 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * $COPYRIGHT
+ *
+ */
+
+#ifndef _DPLASMAAAUX_HIP_H_
+#define _DPLASMAAAUX_HIP_H_
+
+#if defined(DPLASMA_HAVE_HIP)
+#include "parsec/mca/device/hip/device_hip.h"
+
+#include <hipblas/hipblas.h>
+
+#define dplasma_hipblas_side(side)                                        \
+    assert( (side == dplasmaRight) || (side == dplasmaLeft) );            \
+    side = (side == dplasmaRight) ? HIPBLAS_SIDE_RIGHT : HIPBLAS_SIDE_LEFT;
+
+
+#define dplasma_hipblas_diag(diag)                                             \
+    assert( (diag == dplasmaNonUnit) || (diag == dplasmaUnit) );               \
+    diag = (diag == dplasmaNonUnit) ? HIPBLAS_DIAG_NON_UNIT : HIPBLAS_DIAG_UNIT;
+
+#define dplasma_hipblas_fill(fill)                                                   \
+    assert( (fill == dplasmaLower) || (fill == dplasmaUpper) );                      \
+    fill = (fill == dplasmaLower) ? HIPBLAS_FILL_MODE_LOWER : HIPBLAS_FILL_MODE_UPPER;
+
+#if defined(PRECISION_z) || defined(PRECISION_c)
+#define dplasma_hipblas_op(trans)                 \
+    assert( (trans == dplasmaNoTrans) || (trans == dplasmaTrans) || (trans == dplasmaConjTrans) ); \
```

Comment:
I really dislike these being macros. Inline functions would be more appropriate and less error prone but if the CUDA variants do the same then so be it.